apps/st2-history/history-panel.component.js

```diff
@@ -157,9 +160,26 @@ export default class HistoryPanel extends React.Component {
       this.setState({ id });
     }
 
-    if (next.page !== current.page || !_.isEqual(next.activeFilters, current.activeFilters)) {
-      this.fetchGroups(next);
+    if (page === current.page && 
+      _.isEqual(activeFilters, current.activeFilters) &&
+      filter === this.props.filter
+    ) {
+      return;
+    }
+
+    const tokens = filter.split(' ');
+    if (!tokens.length) {
+      return;
+    }
+
+    for (const token of tokens) {
+      const [ , v ] = token.split(':');
+      if (!v) {
+        return;
+      }
     }
```

Comment:
should we add a comment for what this is looking for? It looks like it will halt requests until the filter has a value, so things like `foo` and `foo:` will be blocked. Might be pertinent to move this logic to a `isFilterValid` method, making it read a little cleaner, especially with future code that might have a more rigorous check here.

Should we add some throttling (somewhere), so that `foo:bar` will not trigger 3 lookups?